enterprise/server/util/redis/redis_test.go

```diff
@@ -0,0 +1,309 @@
+package redis_test
+
+import (
+	"crypto/tls"
+	"fmt"
+	"net"
+	"strconv"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+
+	redisutil "github.com/buildbuddy-io/buildbuddy/enterprise/server/util/redis"
+	redis "github.com/go-redis/redis/v8"
+)
+
+func genRedisOptions(
+	protocol string,
+	user string,
+	password string,
+	hostname string,
+	port string,
+	database string,
+) *redis.Options {
```

Comment:
One nifty thing you can do in go when you have many arguments of the same type is omit the type on all but the last of that kind. So this could be:

func genRedisOptions(protocol, user, password, hostname, port, database string) *redis.Options